workflow/local_docker_build.sh

```diff
@@ -0,0 +1,13 @@
+#!/bin/sh
+# This script can be used to locally check and debug
+# the linux build process outside of CI.
+# This should be run from the root directory `packaging/manual_docker_build.sh`
```

Comment:
I'm not very familiar with Docker, but on Ubuntu 16.10 you can only run `docker` commands as root. Since this script appears to build from the root of the source tree, this leaves a bunch of root-owned files sprinkled around the source tree. Can we instead build from a directory within the docker image, or otherwise outside of the source tree, to avoid this contamination?